projects/packages/plugins-installer/src/class-jetpack-automatic-install-skin.php

```diff
@@ -10,55 +18,81 @@
 class Jetpack_Automatic_Install_Skin extends Automatic_Upgrader_Skin {
 	/**
 	 * Stores the last error key;
+	 *
+	 * @var string
 	 **/
 	protected $main_error_code = 'install_error';
 
 	/**
 	 * Stores the last error message.
+	 *
+	 * @var string
 	 **/
 	protected $main_error_message = 'An unknown error occurred during installation';
 
 	/**
 	 * Overwrites the set_upgrader to be able to tell if we e ven have the ability to write to the files.
 	 *
-	 * @param WP_Upgrader $upgrader
-	 *
+	 * @param WP_Upgrader $upgrader The upgrader object.
 	 */
 	public function set_upgrader( &$upgrader ) {
 		parent::set_upgrader( $upgrader );
 
 		// Check if we even have permission to.
 		$result = $upgrader->fs_connect( array( WP_CONTENT_DIR, WP_PLUGIN_DIR ) );
 		if ( ! $result ) {
-			// set the string here since they are not available just yet
+			// set the string here since they are not available just yet.
 			$upgrader->generic_strings();
 			$this->feedback( 'fs_unavailable' );
 		}
 	}
 
 	/**
 	 * Overwrites the error function
+	 *
+	 * @param \WP_Error|mixed $error The error object.
 	 */
 	public function error( $error ) {
 		if ( is_wp_error( $error ) ) {
 			$this->feedback( $error );
 		}
 	}
 
+	/**
+	 * Don't set the process_failed as code since it is not that helpful unless we don't have one already set
+	 *
+	 * @param string $code The error code.
+	 * @return void
+	 */
 	private function set_main_error_code( $code ) {
-		// Don't set the process_failed as code since it is not that helpful unless we don't have one already set.
-		$this->main_error_code = ( $code === 'process_failed' && $this->main_error_code ? $this->main_error_code : $code );
+		$this->main_error_code = ( 'process_failed' === $code && $this->main_error_code ? $this->main_error_code : $code );
 	}
 
+	/**
+	 * Don't set the process_failed as message since it is not that helpful unless we don't have one already set
+	 *
+	 * @param string $message The error message.
+	 * @param string $code The error code.
+	 * @return void
+	 */
 	private function set_main_error_message( $message, $code ) {
-		// Don't set the process_failed as message since it is not that helpful unless we don't have one already set.
-		$this->main_error_message = ( $code === 'process_failed' && $this->main_error_code ? $this->main_error_code : $message );
+		$this->main_error_message = ( 'process_failed' === $code && $this->main_error_code ? $this->main_error_code : $message );
```

Comment:
I think the code is right. It checks for the error code but sets the error message.

Reply:
I haven't tried it, but it looks to me that with the current code you'd get the following behavior:
```php
$instance->set_main_error_code( "foo" );
$instance->set_main_error_message( "Message for foo", "foo" );
echo $instance->get_main_error_message() . "\n"; // Prints "Message for foo".
$instance->set_main_error_message( "Unused message", "process_failed" );
echo $instance->get_main_error_message() . "\n"; // Now prints "foo", not "Message for foo".
```
Whether we should check the code or message in the first part is debatable (personally I'd probably merge `set_main_error_code` and `set_main_error_message` into one method to avoid the ambiguity), but the second should certainly not be the code.

Reply:
Yeah I get your point now. I'm not sure if we should change the second check against the existence of the code or the message...

I went ahead and merged the PRto unblock the work and will prepare a new one to address this.

Worth saying that this has been there for many many years, I just moved the code around.